packages/nova-react/src/eventing/nova-eventing-provider.tsx

```diff
@@ -142,36 +154,63 @@ export const NovaEventingInterceptor: React.FunctionComponent<
     interceptorRef.current = interceptor;
   }
 
-  const { internal } = React.useContext(NovaEventingContext);
+  const { internal: rootInternal } = React.useContext(NovaEventingContext);
 
-  if (!internal) {
+  if (!rootInternal) {
     invariant(
-      internal,
+      rootInternal,
       "Nova Eventing provider must be initialized prior to creating NovaEventingInterceptor!",
     );
   }
 
-  const reactEventing = React.useMemo(
-    generateEventing(internal.eventingRef, internal.mapperRef, interceptorRef),
+  const reactEventing: NovaReactEventing = React.useMemo(
+    () => ({
+      bubble: async (eventWrapper: ReactEventWrapper) => {
+        const mappedEvent: EventWrapper =
+          rootInternal.mapperRef.current(eventWrapper);
+        return getBubble(rootInternal.eventingRef, mappedEvent, interceptorRef);
+      },
+      generateEvent: getGenerateEvent(rootInternal.eventingRef, interceptorRef),
+    }),
     [],
   );
 
-  const reactUnmountEventing = React.useMemo(
-    generateEventing(
-      internal.unmountEventingRef,
-      internal.mapperRef,
-      interceptorRef,
-    ),
+  const reactUnmountEventing: NovaReactEventing = React.useMemo(
+    () => ({
+      bubble: async (eventWrapper: ReactEventWrapper) => {
+        const mappedEvent: EventWrapper =
+          rootInternal.mapperRef.current(eventWrapper);
+        return getBubble(
+          rootInternal.unmountEventingRef,
+          mappedEvent,
+          interceptorRef,
+        );
+      },
+      generateEvent: getGenerateEvent(
+        rootInternal.unmountEventingRef,
+        interceptorRef,
+      ),
+    }),
     [],
   );
 
+  // Internal should point to eventing/unmountEventing created by the interceptor
```

Comment:
How about `Internal should point to eventing/unmountEventing created by the interceptor, so that we can nest arbitrary numbers of interceptors`?